src/register-org/components/regulator-details/regulator-details.component.ts

```diff
@@ -68,7 +72,12 @@ export class RegulatorDetailsComponent extends RegisterComponent implements OnIn
       }
       default: {
         formGroup.removeControl('regulatorName');
-        formGroup.addControl('organisationRegistrationNumber', new FormControl(formGroup.value.organisationRegistrationNumber, Validators.required));
+        if (formGroup.get('organisationRegistrationNumber')) {
+          formGroup.get('organisationRegistrationNumber').reset();
+        } else {
+          formGroup.addControl('organisationRegistrationNumber', new FormControl(null, Validators.required));
```

Comment:
Minor comment - organisationRegistrationNumber is a long string to repeat six times - could we put this in a string constant/enum?